helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/ClusterContext.java

```diff
@@ -135,6 +140,10 @@ public float getEstimatedMaxUtilization() {
     return _estimatedMaxUtilization;
   }
 
+  public ClusterModelProvider.RebalanceScopeType getScopeType() {
```

Comment:
As we discussed offline, this type will break the module design. It introduces a detail in one component to the other one. It would be better to make the constraint algorithm independent.
Please have a try to split the constraint in the other way, baseline check constraint vs. best possible check constrant. Then we can tune the constraint weight to achieve our goal.